src/ServiceControl/EventLog/EventLogMappings.cs

```diff
@@ -13,12 +13,12 @@ public EventLogMappings(Dictionary<Type, Type> mappings)
             this.mappings = mappings;
         }
 
-        public bool HasMapping(IEvent message)
+        public bool HasMapping(IDomainEvent message)
         {
             return mappings.ContainsKey(message.GetType());
         }
 
-        public EventLogItem ApplyMapping(string messageId, IEvent @event)
+        public EventLogItem ApplyMapping(string messageId, IDomainEvent @event)
         {
             var mapping = (IEventLogMappingDefinition)Activator.CreateInstance(mappings[@event.GetType()]);
```

Comment:
Not relevant here but we should actually make sure we don't use `Activator.CreateInstance` on a hot path.

Reply:
raised https://github.com/Particular/ServiceControl/issues/1109